src/diagnostics/netcdf_writer.jl

```diff
@@ -534,21 +590,39 @@ function write_field!(
 
     # TODO: Expand this once we support spatial reductions
     if !haskey(writer.remappers, var.short_name)
+
+        # hpts, vpts are ranges of numbers
+        # hcoords, zcoords are ranges of Geometry.Points
+
+        zcoords = []
+
         if is_horizontal_space
             hpts = target_coordinates(space, writer.num_points)
             vpts = []
         else
             hpts, vpts = target_coordinates(space, writer.num_points)
         end
 
-        # zcoords is going to be empty for a 2D horizontal slice
-        zcoords = [Geometry.ZPoint(p) for p in vpts]
         hcoords = hcoords_from_horizontal_space(
             horizontal_space,
             Meshes.domain(horizontal_space.topology),
             hpts,
         )
 
+        # When we disable vertical_interpolation, we override the vertical points with
+        # the reference values for the vertical space.
+        if writer.disable_vertical_interpolation && !is_horizontal_space
+            # We need Array(parent()) because we want an array of values, not a DataLayout
+            # of Points
+            vpts = Array(
+                parent(
+                    space.grid.vertical_grid.center_local_geometry.coordinates,
+                ),
+            )
```

Comment:
@simonbyrne is there a better way to do this?

I need a simple array of numbers

Reply:
This works.

We probably should define an accessor function for getting the "reference" vertical grid.

Reply:
I agree.